### PR TITLE
🐛 [Story interactive] Fix Post-Selection Animation Inaccuracies for Image Quizzes

### LINKED ISSUE #35287: [Story interactive] White Flash on Selection for Dark Theme Image Quizzes
### Description

On some browsers, there is a flash of white around the answer choice bubble (with a checkmark or cross) of the selected option when using a dark theme image quiz

https://user-images.githubusercontent.com/20510288/126342623-35d6bcac-9d30-47ee-ad55-9fc039ea791d.mp4

For the above video, notice the white border flash around the option B answer choice bubble on the first page and the same around the option A answer choice bubble on the second page

### Reproduction Steps

On an affected browser, navigate to a dark theme image quiz component and select any option

### Relevant Logs

_No response_

### Browser(s) Affected

Chrome, Safari

### OS(s) Affected

Chrome and Safari for iOS, Safari for macOS

### Device(s) Affected

_No response_

### AMP Version Affected

_No response_

### PR DESCRIPTION
**Pull request by Brandons42**
Removes a flash of white around the answer choice bubble of the selected option that appears on some browsers when using a dark theme image quiz and slight 1 pixel layout shifts after post-selection animations

Demos:
https://interactive-img.web.app/animation-fix/quiz.html
https://interactive-img.web.app/animation-fix/text-quiz.html

Fixes #35287

/cc @mszylkowski 
/cc @processprocess

### CODE REVIEW

**Comment by mszylkowski** [comment]
What's the issue? Is it that if the number of pixels is not rounded then the rendering glitches?

If so, can you link a "before" demo (any link that doesn't have this fix) so we can cross check the fix works?

**Comment by Brandons42** [comment]
> What's the issue? Is it that if the number of pixels is not rounded then the rendering glitches?
> 
> If so, can you link a "before" demo (any link that doesn't have this fix) so we can cross check the fix works?

I updated the issue (#35287) with a demo video that was made at this link on Chrome for iOS: https://interactive-img.web.app/fill-bug/quiz.html

File: extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.css [hunk 1]
```diff
@@ -15,11 +15,13 @@
...
 
 .i-amphtml-story-interactive-img-quiz-answer-choice {
+  --i-amphtml-interactive-answer-choice-border: 0.125em solid
+    var(--i-amphtml-interactive-option-background-color) !important;
   position: absolute !important;
   height: 1.5em !important;
   width: 1.5em !important;
   bottom: 0.5em !important;
-  border: 0.125em solid var(--i-amphtml-interactive-option-background-color) !important;
+  border: var(--i-amphtml-interactive-answer-choice-border) !important;
```
Comment (processprocess) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.css:24]: Would this be fixed if we remove line 24? I'm not sure if it's still useful.

That way we wouldn't need to override [--i-amphtml-interactive-answer-choice-border](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/amp-story.css#L914)

**Comment by Brandons42** [comment]
Just removed the border style rule from the keyframes animation, demos for the image and text quiz are in the description

**Comment by mszylkowski** [comment]
I see on the text quiz demo that the check/cross icons in the animation shift when the animation ends, which I think could be coming from the border change. You may need an extra rule somewhere that adds the border directly to the circle

File: extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.css [hunk 2]
```diff
@@ -19,14 +19,15 @@
...
-  border: 0.125em solid var(--i-amphtml-interactive-option-background-color) !important;
+  box-shadow: 0 0 0 0.125em var(--i-amphtml-interactive-option-background-color) !important;
   background-color: var(--interactive-accent-color) !important;
   border-radius: 50% !important;
   text-align: center !important;
   color: var(--i-amphtml-interactive-option-answer-choice-color) !important;
   font-weight: 700 !important;
   line-height: 1.5em !important;
   z-index: 1 !important;
+  backface-visibility: hidden !important;
```
Comment (processprocess) [inline extensions/amp-story-interactive/0.1/amp-story-interactive-img-quiz.css:30]: Please add a comment above this. 
Could be something describing the behavior like "Prevents 1 pixel shift on animation complete".

**Comment by mszylkowski** [comment]
The ABCD circle borders are cut-off in some cases due to browser glitches, adding a margin will make it go away in some cases but the glitch still persists sometimes. Let us know when that's fixed, but the other stuff LGTM

**Review by mszylkowski — APPROVED** [review]

**Review by processprocess — APPROVED** [review]

**Pull request merged**